packages/@aws-cdk/aws-stepfunctions-tasks/test/emrcontainers/start-job-run.test.ts

```diff
@@ -2,594 +2,766 @@
+    });
+
+    // THEN
+    expect(stack.resolve(task.toStateJson())).toMatchObject({
+      Parameters: {
+        VirtualClusterId: clusterId,
+        ReleaseLabel: ReleaseLabel.EMR_6_2_0.label,
+        JobDriver: {
+          SparkSubmitJobDriver: {
+            EntryPoint: 'local:///usr/lib/spark/examples/src/main/python/pi.py',
+            SparkSubmitParameters: '--conf spark.executor.instances=2',
+          },
         },
-      },
-      ConfigurationOverrides: {
-        MonitoringConfiguration: {
-          CloudWatchMonitoringConfiguration: {
-            LogGroupName: {
-              Ref: 'EMRContainersStartJobRunMonitoringLogGroup882D450C',
+        ConfigurationOverrides: {
+          MonitoringConfiguration: {
+            CloudWatchMonitoringConfiguration: {
+              LogGroupName: {
+                Ref: 'EMRContainersStartJobRunMonitoringLogGroup882D450C',
+              },
             },
-          },
-          PersistentAppUI: 'ENABLED',
-          S3MonitoringConfiguration: {
-            LogUri: {
-              'Fn::Join': [
-                '',
-                [
-                  's3://',
-                  {
-                    Ref: 'EMRContainersStartJobRunMonitoringBucket8BB3FC54',
-                  },
+            PersistentAppUI: 'ENABLED',
+            S3MonitoringConfiguration: {
+              LogUri: {
+                'Fn::Join': [
+                  '',
+                  [
+                    's3://',
+                    {
+                      Ref: 'EMRContainersStartJobRunMonitoringBucket8BB3FC54',
+                    },
+                  ],
                 ],
-              ],
+              },
             },
           },
         },
+        ExecutionRoleArn: {
+          'Fn::GetAtt': [
+            'EMRContainersStartJobRunJobExecutionRole40B8DD81',
+            'Arn',
+          ],
+        },
       },
-      ExecutionRoleArn: {
-        'Fn::GetAtt': [
-          'EMRContainersStartJobRunJobExecutionRole40B8DD81',
-          'Arn',
-        ],
-      },
-    },
+    });
   });
-});
 
-test('Invoke EMR Containers Start Job Run with Job Execution Role', () => {
+  test('provided from user', () => {
+    // WHEN
 
-  // WHEN
-  const task = new EmrContainersStartJobRun(stack, 'EMR Containers Start Job Run', {
-    virtualClusterId: sfn.TaskInput.fromText(clusterId),
-    releaseLabel: ReleaseLabel.EMR_6_2_0,
-    executionRole: iam.Role.fromRoleArn(stack, 'Job Execution Role', 'arn:aws:iam::xxxxxxxxxxxx:role/JobExecutionRole'),
-    jobDriver: {
-      sparkSubmitJobDriver: {
-        entryPoint: sfn.TaskInput.fromText('local:///usr/lib/spark/examples/src/main/python/pi.py'),
-        sparkSubmitParameters: '--conf spark.executor.instances=2 --conf spark.executor.memory=2G --conf spark.executor cores=2 --conf spark.driver.cores=1',
+    const logGroup = new logs.LogGroup(stack, 'Log Group');
+    const s3Bucket = new s3.Bucket(stack, 'Bucket');
```

Comment:
use `LogGroup.fromLogGroupName` and `Bucket.fromBucketName` instead so we can confirm that the synthesized definitions are using the right ones (which doesn't seem to be the case right now, they look generated)